src/main/java/org/terasology/caves/CaveToSurfaceProvider.java

```diff
@@ -47,7 +48,7 @@ public void process(GeneratingRegion region) {
         SurfacesFacet surfacesFacet = region.getRegionFacet(SurfacesFacet.class);
         SeaLevelFacet seaLevel = region.getRegionFacet(SeaLevelFacet.class);
 
-        Set<Vector3i> cavePositions = new HashSet<Vector3i>();
+        Set<Vector3ic> cavePositions = Sets.newHashSetWithExpectedSize(caveFacet.getWorldRegion().getSizeX() * caveFacet.getWorldRegion().getSizeZ());
```

Comment:
Why this formula for the expected size? Some fraction of the volume would make more sense.

Reply:
Its probably oversized. umm what fraction of the volume would be best?

Reply:
I agree that using a fraction of the region's volume would express the intent better here.

Just a wild guess: I guess we would not expect more than 50% to be cave positions? Maybe even less, so maybe 1/3? The set will grow if we add more positions, so we can play around with the value to fine tune that. 

```suggestion
        Set<Vector3ic> cavePositions = Sets.newHashSetWithExpectedSize(0.33f * caveFacet.getWorldRegion().volume());
```